src/main/java/com/azure/spring/dev/tools/actions/UpdateSpringCloudAzureSupportFileRunner.java

```diff
@@ -115,13 +115,18 @@ private String findCompatibleSpringCloudVersion(String springBootVersion) {
         if (supportMetadata != null && supportMetadata.getSupportStatus() == SupportStatus.END_OF_LIFE) {
             return supportMetadata.getSpringCloudVersion();
         }
-        return this.springCloudCompatibleSpringBootVersionRanges
-            .entrySet()
-            .stream()
-            .filter(entry -> entry.getValue().match(Version.parse(springBootVersion)))
-            .map(Map.Entry::getKey)
-            .collect(Collectors.toList())
-            .stream().findFirst().orElseThrow();
+        try {
+            return this.springCloudCompatibleSpringBootVersionRanges
+                .entrySet()
+                .stream()
+                .filter(entry -> entry.getValue().match(Version.parse(springBootVersion)))
+                .map(Map.Entry::getKey)
+                .collect(Collectors.toList())
+                .stream().findFirst().orElseThrow();
```

Comment:
If you are catching the exception here, how about not throwing it in the first place? And what's the caller of the function going to do with the string "not get the supported Spring Cloud version yet" you return?